samcli/lib/hook/hook_wrapper.py

```diff
@@ -179,3 +177,20 @@ def _execute_as_module(module: str, method: str, params: Optional[Dict] = None)
 
     result = getattr(mod, method)(params)
     return cast(Dict, result)
+
+
+def get_available_hook_packages_ids() -> List[str]:
+    """
+    return a list of available hook packages ids.
+
+    Returns
+    -------
+    List
+        The available hook packages ids.
+    """
+    LOG.debug("Return available internal hook packages")
+    hook_packages_ids = []
+    for child in _INTERNAL_PACKAGES_ROOT.iterdir():
```

Comment:
Q: does this mean if the appropriate folder structure is adhered to, users can additional hooks that sam cli will recognize?

Reply:
actually, this is the way we used to implement our own hooks, but for external hooks implemented by community, we are going to have a new command to install these hooks, and the plan is to maintain the list of available hooks in some configuration file.